src/navigation/navigation_main.cc

```diff
@@ -80,25 +103,79 @@ CONFIG_FLOAT(laser_loc_x, "NavigationParameters.laser_loc.x");
 CONFIG_FLOAT(laser_loc_y, "NavigationParameters.laser_loc.y");
 
 DEFINE_string(map, "UT_Campus", "Name of navigation map file");
+DEFINE_string(twist_drive_topic, "navigation/cmd_vel", "Drive Command Topic");
 
-DECLARE_int32(v);
+// DECLARE_int32(v);
 
 bool run_ = true;
+bool simulate_ = false;
+bool enabled_ = false;
+bool received_odom_ = false;
+bool received_laser_ = false;
+bool goal_set_ = false;
+Vector2f goal_ = {0, 0};
+Vector2f current_loc_ = {0, 0};
+Vector2f current_vel_ = {0, 0};
+float current_angle_ = 0;
+float goal_angle_ = 0;
+navigation::Odom odom_;
+vector<Vector2f> point_cloud_;
 sensor_msgs::LaserScan last_laser_msg_;
 Navigation navigation_;
 
+// Publishers
+ros::Publisher ackermann_drive_pub_;
+ros::Publisher vis_pub_;
+ros::Publisher twist_drive_pub_;
+ros::Publisher viz_pub_;
+ros::Publisher map_lines_publisher_;
+ros::Publisher pose_marker_publisher_;
+ros::Publisher nav_status_pub_;
+ros::Publisher status_pub_;
+ros::Publisher fp_pcl_pub_;
+ros::Publisher path_pub_;
+ros::Publisher carrot_pub_;
+
+// Messages
+visualization_msgs::Marker line_list_marker_;
+visualization_msgs::Marker pose_marker_;
+visualization_msgs::Marker target_marker_;
+VisualizationMsg local_viz_msg_;
+VisualizationMsg global_viz_msg_;
+
 void EnablerCallback(const std_msgs::Bool& msg) {
-  navigation_.Enable(msg.data);
+  enabled_ = msg.data;
 }
 
-void LaserCallback(const sensor_msgs::LaserScan& msg) {
+navigation::Odom OdomHandler(const nav_msgs::Odometry& msg) {
+  if (FLAGS_v > 0) {
+    printf("Odometry t=%f\n", msg.header.stamp.toSec());
+  }
+  navigation::Odom odom;
+  odom.time = msg.header.stamp.toSec();
+  odom.orientation = {static_cast<float>(msg.pose.pose.orientation.w),
+                      static_cast<float>(msg.pose.pose.orientation.x),
+                      static_cast<float>(msg.pose.pose.orientation.y),
+                      static_cast<float>(msg.pose.pose.orientation.z)};
+  odom.position = {static_cast<float>(msg.pose.pose.position.x),
+                   static_cast<float>(msg.pose.pose.position.y),
+                   static_cast<float>(msg.pose.pose.position.z)};
+  return odom;
+}
+
+void OdometryCallback(const nav_msgs::Odometry& msg) {
+  received_odom_ = true;
+  odom_ = OdomHandler(msg);
+}
+
+vector<Vector2f> LaserHandler(const sensor_msgs::LaserScan& msg) {
   if (FLAGS_v > 0) {
     printf("Laser t=%f, dt=%f\n",
            msg.header.stamp.toSec(),
            GetWallTime() - msg.header.stamp.toSec());
   }
   // Location of the laser on the robot. Assumes the laser is forward-facing.
-  const Vector2f kLaserLoc(CONFIG_laser_loc_x, CONFIG_laser_loc_y);
+  const Vector2f kLaserLoc(0.065, 0);
```

Comment:
This should not be hard-coded - please revert to previous version with the config variables.